semgrep/semgrep/app/session.py

```diff
@@ -31,13 +33,43 @@ class UserAgent:
 
     @tags.default
     def get_default_tags(self) -> Set[str]:
+        result = set()
         if os.getenv("SEMGREP_USER_AGENT_APPEND"):
-            return set({os.environ["SEMGREP_USER_AGENT_APPEND"]})
-        return set()
+            result.add(os.environ["SEMGREP_USER_AGENT_APPEND"])
+
+        try:
+            remote_url = subprocess.check_output(
+                ["git", "remote", "get-url", "origin"],
+                cwd=Path(__file__).parent,
+                stderr=subprocess.DEVNULL,
+                encoding="utf-8",
+            ).strip()
+            sha = subprocess.check_output(
+                [
+                    "git",
+                    "describe",
+                    # a --match value never matches will give us SHA instead of most recent tag
+                    "--match=nah_dont_actually_match",
+                    "--always",
+                    "--dirty",
+                ],
+                cwd=Path(__file__).parent,
+                stderr=subprocess.DEVNULL,
+                encoding="utf-8",
+            ).strip()
+            # If we installed semgrep in `.venv/` in the semgrep-docs repo,
+            # this function would report semgrep-docs commit SHAs in the user agent.
+            # This is why we verify the origin URL, and why check with .endswith()
+            if remote_url.replace(".git", "").endswith("/semgrep"):
+                result.add(f"sha/{sha}")
+        except subprocess.CalledProcessError:
+            pass
+
+        return result
 
     def __str__(self) -> str:
         result = f"{self.name}/{self.version}"
-        for note in sorted(self.tags):
+        for note in sorted(self.tags, key=lambda x: ("/" in x, x)):
```

Comment:
is note supposed to be a (Tuple, str)

Reply:
nope, this is just the sorting key; essentially what this does is:

1. first, order all notes to the front that have no `/` in them
2. then, order the rest alphabetically

Reply:
so this leads to this ordering: `(Docker) (command/scan) (sha/123987)`

Reply:
<img width="351" alt="CleanShot 2022-05-24 at 15 39 20@2x" src="https://user-images.githubusercontent.com/1060436/170143439-7fbacb1a-94a8-4650-97c4-711987ffc2ac.png">

sample in practice

Reply:
I see